File: GOF/Builder/cpp/director/MazeGame.cpp
#include "MazeGame.h"
#include "../builder/MazeBuilder.h"

std::shared_ptr<Maze> MazeGame::createMaze(std::weak_ptr<MazeBuilder> builder)
{
	builder.lock()->buildMaze();
	builder.lock()->buildRoom(1);
	builder.lock()->buildRoom(2);
	builder.lock()->buildDoor(1, 2);

	return builder.lock()->getMaze();
}

File: GOF/Builder/java/src/director/MazeGame.java
package director;

import Builder.MazeBuilder;
import Maze.Maze;

public class MazeGame {
	public Maze createMaze(MazeBuilder builder) {
		builder.buildMaze();

		builder.buildRoom(1);
		builder.buildRoom(2);

		builder.buildDoor(1, 2);

		return builder.getMaze();
	}
}
